env0/resource_environment.go

```diff
@@ -228,6 +228,15 @@ func resourceEnvironment() *schema.Resource {
 				},
 			},
 		},
+
+		CustomizeDiff: customdiff.ForceNewIf("template_id", func(ctx context.Context, d *schema.ResourceDiff, meta interface{}) bool {
+			// For templateless: any changes in template_id, no need to do anything (template id can't change).
+			// This is done due to historical bugs/issues.
+			if _, ok := d.GetOk("without_template_settings.0"); ok {
+				return false
+			}
+			return true
```

Comment:
If the values don't change it ignores the "true" - this is exactly what we want.